include/envoy/http/header_map.h

```diff
@@ -54,6 +54,15 @@ class LowerCaseString {
     lower();
   }
 
+  LowerCaseString& operator=(LowerCaseString&& rhs) {
```

Comment:
Would prefer to not add these as they can lead to magic C++ copies that are hard to diagnose. Can you remove and find a different way to do what you need?

Reply:
@mattklein123 would just move assignment be fine here? This doesn't seem like it leads to performance considerations.

Reply:
For this type of perf critical stuff my preference is to not have any magic methods which are hard to reason about, but sure, move assignment seems fine. Is this just needed for tests though?

Reply:
@mattklein123 It wouldn't just be for tests, it'd generally be how you populate `DebugConfig`. I can definitely get by without it, though.